tests/ludwig/config_validation/test_validate_config_encoder.py

```diff
@@ -0,0 +1,58 @@
+import pytest
+
+from ludwig.config_validation.validation import validate_config
+from ludwig.constants import DEFAULTS, ENCODER, INPUT_FEATURES, OUTPUT_FEATURES, SEQUENCE, TEXT, TIMESERIES, TYPE
+from ludwig.error import ConfigValidationError
+from tests.integration_tests.utils import (
+    binary_feature,
+    number_feature,
+    sequence_feature,
+    text_feature,
+    timeseries_feature,
+)
+
+
+@pytest.mark.parametrize("feature_type", [SEQUENCE, TEXT, TIMESERIES])
+def test_default_transformer_encoder(feature_type):
+    """Tests that a transformer hyperparameter divisibility error is correctly recognized in feature defaults.
+
+    Transformers require that `hidden_size % num_heads == 0`. 9 and 18 were selected as test values because they were
+    the values from the original error.
+    """
+    config = {
+        INPUT_FEATURES: [number_feature(), sequence_feature(), text_feature(), timeseries_feature()],
+        OUTPUT_FEATURES: [binary_feature()],
+        DEFAULTS: {feature_type: {ENCODER: {TYPE: "transformer", "hidden_size": 9, "num_heads": 18}}},
+    }
+
+    with pytest.raises(ConfigValidationError):
+        validate_config(config)
```

Comment:
Use `ModelConfig.from_dict(config)` instead of `validate_config`, which I'm planning on removing in https://github.com/ludwig-ai/ludwig/pull/3104

Reply:
So all of the tests fail when I use `ModelConfig.from_dict`. Do we need to wait for the PR to land before updating?

Reply:
@jeffkinnison you're totally right! Let me try to land that PR first.